tests/phpunit/CRM/Contribute/Form/ContributionTest.php

```diff
@@ -905,21 +942,74 @@ public function testReSubmitSaleTax() {
         'return' => array('tax_amount', 'total_amount', 'net_amount', 'financial_type_id', 'receive_date', 'payment_instrument_id'),
       )
     );
-    $this->assertEquals(110, $contribution['total_amount']);
-    $this->assertEquals(10, $contribution['tax_amount']);
-    $this->assertEquals(110, $contribution['net_amount']);
+    $this->assertEquals(220, $contribution['total_amount']);
+    $this->assertEquals(20, $contribution['tax_amount']);
+    $this->assertEquals(220, $contribution['net_amount']);
 
     $strings = array(
-      'Total Tax Amount : $ 10.00',
-      'Total Amount : $ 110.00',
+      'Total Tax Amount : $ 20.00',
+      'Total Amount : $ 220.00',
       'Date Received: April 21st, 2015',
       'Paid By: Check',
       'Check Number: 12345',
     );
 
     $mut->checkMailLog($strings);
-    $this->callAPISuccessGetCount('FinancialTrxn', array(), 3);
-    $this->callAPISuccessGetCount('FinancialItem', array(), 2);
+    $this->callAPISuccessGetCount('FinancialTrxn', array(), 4);
+    $items = $this->callAPISuccess('FinancialItem', 'get', array('sequential' => 1));
+    $this->assertEquals(4, $items['count']);
+    $this->assertEquals('Contribution Amount', $items['values'][0]['description']);
+    $this->assertEquals('Sales Tax', $items['values'][1]['description']);
+    $this->assertEquals('Contribution Amount', $items['values'][0]['description']);
+    $this->assertEquals('Sales Tax', $items['values'][1]['description']);
+
+    $this->assertEquals(100, $items['values'][0]['amount']);
+    $this->assertEquals(10, $items['values'][1]['amount']);
+    // @todo what should the amount BE? I believe this is incorrect elsewhere too.
```

Comment:
Pradeep, could you review this. I think we need a test for 
$this->assertEquals(100, $items['values'][3]['amount']);
As we have added 100 to the main value and 10 to the tax amount. $120 seems very wrong to me.